### ISSUE TITLE
BUG: lua-prepend-path and lua-load re-ordering

### ISSUE THREAD

**Issue by c-nv-s** [haproxytech/dataplaneapi#199]
_ _ _ _ _
system info: ubuntu 18.04
HAProxy version 2.4.1-1ppa1~bionic 2021/06/17 

HAProxy Data Plane API v2.3.0 f1bcacc
Build from: git@github.com:haproxytech/dataplaneapi.git
Build date: 2021-05-11T09:33:20Z
_ _ _ _ _

When reformatting the haproxy.cfg file, the dataplaneapi re-orders the ``lua-prepend-path`` and ``lua-load`` entries ( so that the lua-prepend-path entry comes AFTER the lua-load entry ) which then causes haproxy to log an error stating that it cannot find the library/dependency ( which is available from the lua-prepend-path entry ).

working from this template [authelia haproxy.cfg](https://github.com/authelia/authelia/blob/master/internal/suites/example/compose/haproxy/haproxy.cfg) you will see that initially the lua-prepend-path is before the lua-load entry.

after the reformatting the entries have been swapped
```
lua-load /etc/haproxy/auth-request.lua
lua-prepend-path /etc/haproxy/?/http.lua
```
this then leads to an error message:

```
level=error msg="error while updating service: %!w(*configuration.ConfError=&{14 err transactionId=cbbe4ed0-d10e-241b-abde-cda7bb500fba 
\nline=10 msg=\"Lua runtime error: /etc/haproxy/auth-request.lua:25: module 'haproxy-lua-http' not found:\"\nmsg=\"parsing [/tmp/haproxy/haproxy.cfg.cbbe4ed0-d10e-241b-abde-cda7bb500fba:54]: 'http-request' expects '
wait-for-handshake', 'use-service', 'send-spoe-group', 'sc-inc-gpc0(*)', 'sc-inc-gpc1(*)', 'sc-set-gpt0(*)', 'do-resolve(*)', 'cache-use', 'add-acl(*)', 'add-header', 'allow', 'auth', 'capture', 'del-acl
(*)', 'del-header', 'del-map(*)', 'deny', 'disable-l7-retry', 'early-hint', 'normalize-uri', 'redirect', 'reject', 'replace-header', 'replace-path', 'replace-pathq', 'replace-uri', 'replace-value', 'retu
rn', 'set-header', 'set-log-level', 'set-map(*)', 'set-method', 'set-mark', 'set-nice', 'set-path', 'set-pathq', 'set-query', 'set-tos', 'set-uri', 'strict-mode', 'tarpit', 'track-sc(*)', 'set-timeout', 
'wait-for-body', 'set-var(*)', 'unset-var(*)', 'set-priority-class', 'set-priority-offset', 'silent-drop', 'set-src', 'set-src-port', 'set-dst', 'set-dst-port', but got 'lua.auth-request'.\"\nmsg=\"Error
(s) found in configuration file /tmp/haproxy/haproxy.cfg.cbbe4ed0-d10e-241b-abde-cda7bb500fba\"})"
```
where the ``http-request`` entry on line 54, which is the equivalent of line 49 in the template:
```
http-request lua.auth-request be_auth_request /api/verify if protected-frontends
```
throws an error when the lua script is called

What should the expected behaviour be? :  
I would think that either the prepend statements should come before the lua-load statements, or the dataplaneapi should just preserve the order that it found within the original file.

**Comment by mjuraga** [comment]
@c-nv-s Thanks for the extensive report, we should change this so that the lua-prepend-path is never after the lua-load statements. 

Unfortunately how our configuration parser works at the time we don't have a way of preserving the ordering or the statements in the haproxy configuration file.

**Comment by c-nv-s** [comment]
@mjuraga is there not a possiblity to give a rough idea if this will be worked on at some point?
It turns out this might end up being a show-stopper for me in the near future.

**Comment by mjuraga** [comment]
@c-nv-s we will definitely include this in our next minor release that should come in the near future

**Comment by mjuraga** [comment]
@c-nv-s  this is included in the new v2.3.3 release.
